cogs/tags.py

```diff
@@ -1,71 +1,144 @@
-from discord.ext import commands
-import discord
-
 import asyncio
-import asyncpg
+from datetime import datetime
 
+import discord
+from config import TAGS_LOG_CHANNEL_ID
+from discord.ext import commands
+
+from .utils.checks import is_admin, is_engineer_check, is_staff
 from .utils.DataBase.tag import Tag
-from .utils.checks import is_engineer_check, is_admin, is_staff
+
+import re
+
+EMOJIS = [
+    "\N{WHITE HEAVY CHECK MARK}",
+    "\N{CROSS MARK}",
+]
+TAG_CREATE_PATTERN = r"\*\*Tag's name:\*\* ((?:.|\n)+)\n\n\*\*Content:\*\* ((?:.|\n)+)"
+TAG_UPADTE_PATTERN = r"\*\*Tag's name:\*\* ((?:.|\n)+)\n\n\*\*Before:\*\* ((?:.|\n)+)\n\n\*\*After:\*\* ((?:.|\n)+)"
```

Comment:
It can change the content, but they still wouldn't be able to have tag with a length higher than 2000.

Reply:
It can be over 2000 characters, say someone creates a tag with a few characters then `\n\n**After:** ` then many characters, the before is still under 2000 characters. However when editing, the before part will contain the `\n\n**After:** ` delimiter, so the regex will match the after to everything after that fake delimiter. If the new tag text is also 2000 characters, the final tag will be the end of the before text, the delimiter and all of the after text so the final new text will be over 2000 characters. 

Example:
```py
before = "...\n\n**After:** ...[1980 more times]"
after = "." * 2000
```
The regex will match the before as `...` and the after as `...[1980 more times]\n\n**After:** ...[2000 times]` which is longer than 2000 and will make the tag crash

Reply:
One solution would be to include a random delimiter in the embed footer and separate the before and after with that, another would be to check that the before matched by the regex is the same as the current tag text and that the after matched by the regex is under 2000 characters.